scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,435 @@
+            report_txt = handler.read()
+
+    output.report = report_txt
+
+
+def update_insights_inventory():
+    """Call insights-client to update insights inventory."""
+    print("Updating system status in Red Hat Insights.")
+    output, returncode = run_subprocess(["/usr/bin/insights-client"])
+
+    if returncode:
+        raise ProcessError(
+            message="Failed to update Insights Inventory by registering the system again. See output the following output: %s"
+            % output,
+            report="insights-client execution exited with code '%s'." % returncode,
+        )
+
+    print("System registered with insights-client successfully.")
+
+
+def reboot_system():
+    print("Rebooting system in 1 minute.")
+    run_subprocess(["/usr/sbin/shutdown", "-r", "1"], wait=False)
+
+
+def main():
+    leapp_upgrade_output = None
+    try:
+        # Exit if not RHEL 7.9 or 8.4
+        dist, version = get_rhel_version()
+        if dist != "rhel" or is_non_eligible_releases(version):
+            raise ProcessError(
+                message="In-place upgrades are supported only on RHEL distributions.",
+                report='Exiting because distribution="%s" and version="%s"'
+                % (dist, version),
+            )
+
+        output = OutputCollector()
+        upgrade_command = ["/usr/bin/leapp", "upgrade", "--report-schema=1.1.0"]
+        rhui_pkgs = setup_leapp(version)
+
+        # Check for RHUI PKGs
+        use_no_rhsm = should_use_no_rhsm_check(len(rhui_pkgs) > 1, upgrade_command)
+        if use_no_rhsm:
+            install_leapp_pkg_corresponding_to_installed_rhui(rhui_pkgs)
+
+        remove_previous_reports()
+        leapp_upgrade_output = execute_upgrade(upgrade_command)
+        print("Leapp upgrade command successfully executed.")
+        parse_results(output)
+    except ProcessError as exception:
+        print(exception.report)
+        output = OutputCollector(
+            status="ERROR",
+            alert=True,
+            error=False,
+            message=exception.message,
+            report=exception.report,
+        )
+        leapp_upgrade_output = None
+    except Exception as exception:
+        print(str(exception))
+        output = OutputCollector(
+            status="ERROR",
+            alert=True,
+            error=False,
+            message="An unexpected error occurred. Expand the row for more details.",
+            report=str(exception),
+        )
+        leapp_upgrade_output = None
+    finally:
+        print("### JSON START ###")
+        print(json.dumps(output.to_dict(), indent=4))
+        print("### JSON END ###")
+        update_insights_inventory()
+
+        if leapp_upgrade_output and REBOOT_GUIDANCE_MESSAGE in leapp_upgrade_output:
+            print("System is ready to be upgraded. Reboot is required.")
+            reboot_system()
+        else:
+            print("System is NOT ready to be upgraded.")
```

Comment:
Is this something that user should see in UI of tasks? If yes then it should probably be handled differently - in the try block and the condition would put it in the OutputCollector somehow.

Reply:
I have changed the final message in output collector when system is ready to be upgraded to show that the system is being upgraded and reboot is scheduled :+1: